mlflow/gateway/providers/gemini.py

```diff
@@ -1,12 +1,242 @@
+import time
 from typing import Any
 
 from mlflow.gateway.config import GeminiConfig, RouteConfig
+from mlflow.gateway.exceptions import AIGatewayException
 from mlflow.gateway.providers.base import BaseProvider, ProviderAdapter
-from mlflow.gateway.providers.utils import send_request
-from mlflow.gateway.schemas import embeddings
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
 
 
 class GeminiAdapter(ProviderAdapter):
+    @classmethod
+    def chat_to_model(cls, payload, config):
+        # Documentation: https://ai.google.dev/api/generate-content
+        # Example payload for the chat API.
+        #
+        # {
+        #     "contents": [
+        #         {
+        #             "role": "user",
+        #             "parts": [
+        #                 {
+        #                     "text": "System: You are an advanced AI Assistant"
+        #                 }
+        #             ]
+        #         },
+        #         {
+        #             "role": "user",
+        #             "parts": [
+        #                 {
+        #                     "text": "Please give the code for addition of two numbers"
+        #                 }
+        #             ]
+        #         }
+        #     ],
+        #     "generationConfig": {
+        #         "temperature": 0.1,
+        #         "topP": 1,
+        #         "stopSequences": ["\n"],
+        #         "candidateCount": 1,
+        #         "maxOutputTokens": 100,
+        #         "topK": 40
+        #     }
+        # }
+        key_mapping = {
```

Comment:
nits: let's define this mapping outside of this class and name it like `GENERATION_CONFIG_KEY_MAPPING`